browser/ui/views/commands/accelerator_service_factory.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/views/commands/accelerator_service_factory.h"
+
+#include <utility>
+
+#include "base/memory/singleton.h"
+#include "brave/browser/ui/views/commands/accelerator_service.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/views/accelerator_table.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "content/public/browser/browser_context.h"
+#include "ui/base/accelerators/accelerator.h"
+
+namespace commands {
+
+// static
+AcceleratorServiceFactory* AcceleratorServiceFactory::GetInstance() {
```

Comment:
please register this factory in `brave::EnsureBrowserContextKeyedServiceFactoriesBuilt` `brave/browser/browser_context_keyed_service_factories.cc`.

Reply:
Out of interest, what does `EnsureBrowserContextKeyedServiceFactoriesBuilt` solve? Won't the factories be constructed the first time someone gets an instance of the service anyway?

Reply:
> Out of interest, what does `EnsureBrowserContextKeyedServiceFactoriesBuilt` solve? Won't the factories be constructed the first time someone gets an instance of the service anyway?

Usually yes, but if you want your service to be constructed on startup (without anyone accessing it directly), then you are required to do this. Here's some info:
https://github.com/chromium/chromium/blob/8cfe860375abb9c58235c8e7c0fdca0f3646ebed/components/keyed_service/content/browser_context_keyed_service_factory.h#L118-L128

It's just a good practice do this for all factories - if you add a factory, just instantiate it in `EnsureBrowserContextKeyedServiceFactoriesBuilt`.